l2geth/rollup/fees/rollup_fee.go

```diff
@@ -140,9 +141,10 @@ func PaysEnough(opts *PaysEnoughOpts) error {
 
 func mulByFloat(num *big.Int, float *big.Float) *big.Int {
 	n := new(big.Float).SetUint64(num.Uint64())
-	n = n.Mul(n, float)
-	value, _ := float.Uint64()
-	return new(big.Int).SetUint64(value)
+	product := n.Mul(n, float)
+	pfloat, _ := product.Float64()
+	rounded := math.Ceil(pfloat)
+	return new(big.Int).SetUint64(uint64(rounded))
 }
```

Comment:
LOL.

Hide this diff from the universe please. BigInts are impossible & I failed you as a reviewer.

Nice.

Duely noted: Always add unit tests that verify that you're not always multiplying by zero.